boards/arm/mimxrt1060_evk/mimxrt1060_evk_pinctrl.dtsi

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Generated by rt_cfg_utils.py on 2022-01-13
+ */
+
+#include <nxp/nxp_imx/rt/mimxrt1062_iomuxc.h>
+
+&pinctrl {
+	lpuart1_default: lpuart1_default {
+		iomuxc_gpio_ad_b0_12_lpuart1_tx {
+			pinmux = <IOMUXC_GPIO_AD_B0_12_LPUART1_TX>;
+			nxp,drive_strength = <0x6>;
+			nxp,hysteresis_enable = <0>;
+			nxp,open_drain = <0>;
+			nxp,pull_keeper_enable = <0x1>;
+			nxp,pull_keeper_select = <0>;
+			nxp,pull_up_down_config = <0>;
+			nxp,slew_rate = <0>;
+			nxp,software_input_on = <0x1>;
+			nxp,speed = <0x2>;
```

Comment:
suggestion: consider defaulting properties to SoC defaults, so that only required fields have to be defined. In many cases, only a few properties need to be adjusted (at least this is the case for ST/GD/Nordic)

Reply:
Is this possible with pinctrl groups? With nodes like ST uses you can define the majority of the overrides at the SoC level, but with pinctrl groups the definition of the pinctrl node is placed within a board specific group. Default groups could be created for each SOC enabled within Zephyr, but that seems like it would deviate from the two pinctrl methods explained in [Zephyr's documentation](https://docs.zephyrproject.org/latest/guides/pinctrl/index.html).

Of course, there's also the possibility that pinctrl nodes like ST uses are the preferred method. If that's the case, let me know and I can rebuild this PR to use that approach.

Reply:
By defaults I mean defaults in the binding file. If set to SoC defaults, then properties resolve to those values if not defined. For example, assume SoC default for `nxp,speed` is `2`, then just set that property to `2` as default so that you don't need to define that for every group unless it has to be changed to another value.

Reply:
Example: https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/pinctrl/gd%2Cgd32-pinctrl-af.yaml

Reply:
I can set a "sane" default for in the binding file, but different pins on RT parts have different SOC level configurations at boot, so some pins would get incorrect defaults assigned.

Reply:
You can also have multiple compatibles if parts differ (already happens in some other places, e.g. ST PLL nodes, you can have `st,stm32f4xx-pll-clock`.